impeller/entity/geometry.cc

```diff
@@ -16,6 +16,46 @@
 
 namespace impeller {
 
+/// Given a convex polyline, create a triangle fan structure.
+std::pair<std::vector<Point>, std::vector<uint16_t>> TessellateConvex(
+    Path::Polyline polyline) {
+  std::vector<Point> output;
+  output.reserve(polyline.points.size() * 3);
+
+  for (auto j = 0u; j < polyline.contours.size(); j++) {
+    Scalar center_x = 0;
+    Scalar center_y = 0;
+    auto [start, end] = polyline.GetContourPointBounds(j);
+    for (auto i = start; i < end; i++) {
+      const auto& point = polyline.points[i];
+      center_x += point.x;
+      center_y += point.y;
+    }
+    center_x /= end - start;
+    center_y /= end - start;
+
+    for (auto i = start + 1; i < end; i++) {
+      const auto& point_a = polyline.points[i - 1];
+      const auto& point_b = polyline.points[i];
+      output.emplace_back(point_a);
+      output.emplace_back(point_b);
+      output.emplace_back(Point(center_x, center_y));
+    }
+    // Some shapes don't seem to self close?
```

Comment:
If you fill a path that is missing an explicit close, you need to add one.

We should just document this behavior in the header doc.

Reply:
I think that when we convert to the polyline we should add this close if it isn't present. Anyway, this is handled below. i'll update the comment

Reply:
If we do it in polyline conversion, we need to make sure to _not_ add a close if it's for a stroke (or if a close is already present).